annotation.go

```diff
@@ -0,0 +1,386 @@
+	}
+	errStreamTagTooLong = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stream tag must be less than 255 characters",
+	}
+	errStreamNameTooLong = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stream name must be less than 255 characters",
+	}
+	errStreamDescTooLong = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stream description must be less than 1024 characters",
+	}
+	errStickerTooLong = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stickers must be less than 255 characters",
+	}
+	errMsgTooLong = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "message must be less than 4096 characters",
+	}
+	errReversedTimes = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "start time must come before end time",
+	}
+	errMissingStreamName = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stream name must be set",
+	}
+	errMissingStreamTagOrId = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "stream tag or id must be set",
+	}
+	errMissingEndTime = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "end time must be set",
+	}
+	errMissingStartTime = &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "start time must be set",
+	}
+)
+
+// Service is the service contract for Annotations
+type AnnotationService interface {
+	// CreateAnnotations creates annotations.
+	CreateAnnotations(ctx context.Context, orgID platform.ID, create []AnnotationCreate) ([]AnnotationEvent, error)
+	// ListAnnotations lists all annotations matching the filter.
+	ListAnnotations(ctx context.Context, orgID platform.ID, filter AnnotationListFilter) (ReadAnnotations, error)
+	// GetAnnotation gets an annotation by id.
+	GetAnnotation(ctx context.Context, orgID, id platform.ID) (*AnnotationEvent, error)
+	// DeleteAnnotations deletes annotations matching the filter.
+	DeleteAnnotations(ctx context.Context, orgID platform.ID, delete AnnotationDeleteFilter) error
+	// DeleteAnnotation deletes an annotation by id.
+	DeleteAnnotation(ctx context.Context, orgID, id platform.ID) error
+	// UpdateAnnotation updates an annotation.
+	UpdateAnnotation(ctx context.Context, orgID, id platform.ID, update AnnotationCreate) (*AnnotationEvent, error)
+
+	// ListStreams lists all streams matching the filter.
+	ListStreams(ctx context.Context, orgID platform.ID, filter StreamListFilter) ([]ReadStream, error)
+	// CreateOrUpdateStream creates or updates the matching stream by name.
+	CreateOrUpdateStream(ctx context.Context, orgID platform.ID, stream Stream) (*ReadStream, error)
+	// UpdateStream updates the stream by the ID.
+	UpdateStream(ctx context.Context, orgID, id platform.ID, stream Stream) (*ReadStream, error)
+	// DeleteStream deletes the stream metadata by name.
+	DeleteStream(ctx context.Context, orgID platform.ID, streamName string) error
+	// DeleteStreamByID deletes the stream metadata by id.
+	DeleteStreamByID(ctx context.Context, orgID, id platform.ID) error
+}
+
+// timeFunc and timeNow are used for running unit tests
+var timeFunc func() time.Time
```

Comment:
Not a huge fan of having this be a global, especially in the root package... would it be a huge pain to refactor the `Validate` methods to take this as an arg in some way?

Reply:
Maybe the `Validate` methods could take a `defaultNow` parameter, and in "real" code the methods could be passed `time.Now`?

Reply:
Refactoring as you suggested wasn't too bad! Should be better than to have a global variable like that. It's a bit more cumbersome but since the timestamps are such an important aspect of annotations it is worth it to be able to inject the dependency and run tests on the validation methods.